### PR TITLE
10_homework_store_with_amount

### PR DESCRIPTION
**Pull request by golovnevs**
Hello teachers! Here is my homework for lesson 10(conditions) - online store with amount.

### CODE REVIEW

File: 10_homework_store_amount/online_store.css [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+.cartMain {
+	display: flex;
+	flex-direction: column;
+	padding: 2em 20vw;
+}
+
+.card {
+	padding: 1em;
+	border: 1px;
+
```
Comment (MrTim) [inline 10_homework_store_amount/online_store.css]: empty line

File: 10_homework_store_amount/online_store.css [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+.soldoutCard {
+	-webkit-filter: blur(5px);
+	-moz-filter: blur(5px);
+	-o-filter: blur(5px);
+	-ms-filter: blur(5px);
+	filter: blur(5px);
+
```
Comment (MrTim) [inline 10_homework_store_amount/online_store.css]: 2 empty lines

File: 10_homework_store_amount/script.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    }
+];
+let inCart = [];
+function bikeListDisplay() {
+    for (let i = 0; i < bikes.length; i++) {
+        bikeCard = document.createElement("div");
+        bikeCard.setAttribute("class", "card");
+        const main = document.getElementById('main');
+
+        if (bikes[i].amount > 1 && bikes[i].amount <= 3) {
```
Comment (MrTim) [inline 10_homework_store_amount/script.js:100]: you can structure this code a bit better:
```js
if (bikes[i].amount > 1 && bikes[i].amount <= 3) {
  bikeCard.innerHTML = ` 
    <h2>${bikes[i].brand} ${bikes[i].model}</h2>
    <img class="bikePic" id="pic_${i}" src="${bikes[i].imgSrc}" />
    <li>Price: ${bikes[i].price} ${bikes[i].currency}</li>
    <li>Year: ${bikes[i].condition.year}</li>
    <li>Mileage: ${bikes[i].condition.mileage} ${bikes[i].condition.mileageUnit}</li>
    <li>Colour: ${bikes[i].colour}</li>
    <li>Engine: ${bikes[i].engineCapacity} cc</li>
    <p>${bikes[i].description}</p>
    <li id="${i}">Avaliable now: ${bikes[i].amount} (almost out of stock)</li>
  `;
} else if (...) {
  ...
} else {
  ...
}
```

File: 10_homework_store_amount/script.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+          <li>Colour: ${bikes[i].colour}</li>
+          <li>Engine: ${bikes[i].engineCapacity} cc</li>
+          <p>${bikes[i].description}</p>
+          <li id="${i}">Avaliable now: ${bikes[i].amount}</li>
+            `;}
+
+        const buttonElement = document.createElement('button');
+        buttonElement.innerText = 'Add to cart';
+        buttonElement.setAttribute("class", "button");
+        if (bikes[i].amount ==0){
```
Comment (MrTim) [inline 10_homework_store_amount/script.js]: spaces around `=` and please adjust the structure of if-else for regardign comment above

File: 10_homework_store_amount/script.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+            inCart.push(bikes[i]);
+            if (bikes[i].amount == 1) {
+                buttonElement.setAttribute("disabled", "true");
+                buttonElement.setAttribute("class", "buttonDisabled");
+                document.getElementById(i).innerHTML= `
+                 <li id=${i}>Out of stock.</li>
+                `;}
+
+            else if (bikes[i].amount > 2 && bikes[i].amount <= 4) {
+                bikes[i].amount = bikes[i].amount-1;
```
Comment (MrTim) [inline 10_homework_store_amount/script.js]: you can do the following in 2 other ways: `amount -= 1` or `amount = --amount`

**Review by MrTim — CHANGES_REQUESTED** [review]
Nice works! Left some comments regarding the structuring

File: 10_homework_store_amount/online_store.css (hunk 1 above)
Comment (golovnevs) [reply 10_homework_store_amount/online_store.css]: Fixed.

File: 10_homework_store_amount/online_store.css (hunk 2 above)
Comment (golovnevs) [reply 10_homework_store_amount/online_store.css]: That's experimental part of code, i dorgot to delete it. I will be more attentive. Thank you!   Fixed.

File: 10_homework_store_amount/script.js (hunk 4 above)
Comment (golovnevs) [reply 10_homework_store_amount/script.js]: Fixed.

File: 10_homework_store_amount/script.js (hunk 5 above)
Comment (golovnevs) [reply 10_homework_store_amount/script.js]: Fixed, thank you!

File: 10_homework_store_amount/script.js (hunk 3 above)
Comment (golovnevs) [reply 10_homework_store_amount/script.js:100]: Fixed and remembered.

**Review by MrTim — APPROVED** [review]
All good now! I have one suggestion to improve:
In bikeListDisplay function you have a lot of duplication in your if-else statement, this structure
```js
  bikeCard.innerHTML = ` 
    <h2>${bikes[i].brand} ${bikes[i].model}</h2>
    <img class="bikePic" id="pic_${i}" src="${bikes[i].imgSrc}" />
    <li>Price: ${bikes[i].price} ${bikes[i].currency}</li>
    <li>Year: ${bikes[i].condition.year}</li>
    <li>Mileage: ${bikes[i].condition.mileage} ${bikes[i].condition.mileageUnit}</li>
    <li>Colour: ${bikes[i].colour}</li>
    <li>Engine: ${bikes[i].engineCapacity} cc</li>
    <p>${bikes[i].description}</p>
    <li id="${i}">Avaliable now: ${bikes[i].amount} (almost out of stock)</li>
`;
```
is almost the same with only one difference - the last sentence inside li tag. You can create a function for ex. getCardLayout with two parameters: data - your bike data and last sentence. So the idea would be to have something like: 
```js
  const availability = Avaliable now: ${bikes[i].amount} (almost out of stock) // or other sentence depends on available amount
  bikeCard.innerHTML = getCardLayout(bikes[i], availability);
```
This would simplify your code.

But if you don't want to do it you can skip it and just merge as it is.

**Pull request merged**